internal/pkg/manifest/env.go

```diff
@@ -102,9 +109,107 @@ type environmentNetworkConfig struct {
 }
 
 type environmentVPCConfig struct {
-	ID      *string              `yaml:"id,omitempty"`
-	CIDR    *IPNet               `yaml:"cidr,omitempty"`
-	Subnets subnetsConfiguration `yaml:"subnets,omitempty"`
+	ID                  *string              `yaml:"id,omitempty"`
+	CIDR                *IPNet               `yaml:"cidr,omitempty"`
+	Subnets             subnetsConfiguration `yaml:"subnets,omitempty"`
+	SecurityGroupConfig securityGroupConfig  `yaml:"security_group,omitempty"`
+}
+
+type securityGroupConfig struct {
+	Ingress []SecurityGroupRule `yaml:"ingress,omitempty"`
+	Egress  []SecurityGroupRule `yaml:"egress,omitempty"`
+}
+
+func (cfg securityGroupConfig) isEmpty() bool {
+	return len(cfg.Ingress) == 0 && len(cfg.Egress) == 0
+}
+
+// SecurityGroupRule holds the security group ingress and egress configs.
+type SecurityGroupRule struct {
+	CidrIP      string      `yaml:"cidr"`
+	PortsConfig PortsConfig `yaml:"ports"`
+	IpProtocol  string      `yaml:"ip_protocol"`
+}
+
+// PortsConfig represents a range of ports [from:to] inclusive.
+// The simple form allow represent from and to ports as a single value, whereas the advanced form is for different values.
+type PortsConfig struct {
+	Port  *int            // 0 is a valid value, so we want the default value to be nil.
+	Ports *PortsRangeBand // Mutually exclusive with port.
+}
```

Comment:
what do you think of these naming:
```go
// SecurityGroupRule holds the security group ingress and egress configs.
type SecurityGroupRule struct {
	CidrIP      string      `yaml:"cidr"`
	Ports        PortsConfig `yaml:"ports"`
	IpProtocol  string      `yaml:"ip_protocol"`
}

// PortsConfig represents a range of ports [from:to] inclusive.
// The simple form allow represent from and to ports as a single value, whereas the advanced form is for different values.
type PortsConfig struct {
	Value  *int            // 0 is a valid value, so we want the default value to be nil.
	Range *PortsRangeBand // Mutually exclusive with port.
}
```

so that when called, it reads
```go
var cfg SecurityGroupRule
cfg.Ports.Value
cfg.Ports.Range
```

Reply:
Yeah this looks clean, I like it. I will do this change.